### ISSUE TITLE
Add autograd tests to verify correctness for C -> R cases

### ISSUE THREAD

**Issue by anjali411** [pytorch/pytorch#52457]
We have been updating the derivative formula in `derivatives.yaml` to return correct gradients for complex functions, however we don't have many tests to verify the correctness of `R -> C` cases. Some of the issues have been reported and fixed (e.g., Fix for `torch.cat` https://github.com/pytorch/pytorch/pull/51681) , however there are probably others that have gone undetected so far.
This is a tracker to systematically test the `R -> C` cases for complex functions in the new `OpInfo` based tests:

- [ ] `torch.add`
- [ ] `torch.sub`
- [ ] `torch.mul`
- [ ] `torch.div`
- [ ] `torch.addcmul`
- [ ] `torch.addcdiv`
- [ ] `torch.cat`

**Comment by anjali411** [comment]
@kshitij12345 would you like to work on this issue?
